ruma-common/src/push/condition.rs

```diff
@@ -42,13 +49,203 @@ pub enum PushCondition {
     },
 }
 
+impl PushCondition {
+    /// Check if this condition applies to the event.
+    ///
+    /// # Arguments
+    ///
+    /// * `event` - A map containing the flattened JSON representation of a room message event.
+    /// * `context` - The context of the room at the time of the event.
+    pub fn applies(
+        &self,
+        event: &BTreeMap<String, String>,
+        context: &PushConditionRoomCtx,
+    ) -> bool {
+        match self {
+            Self::EventMatch { key, pattern } => {
+                let value = match key.as_str() {
+                    "room_id" => &context.room_id,
+                    _ => match event.get(key) {
+                        Some(v) => v,
+                        None => return false,
+                    },
+                };
+
+                matches_pattern(value, pattern, key == "content.body")
+            }
+            Self::ContainsDisplayName => {
+                let value = match event.get("content.body") {
+                    Some(v) => v,
+                    None => return false,
+                };
+
+                matches_pattern(value, &context.user_display_name, true)
+            }
+            Self::RoomMemberCount { is } => is.contains(&context.member_count),
+            Self::SenderNotificationPermission { key } => {
+                let sender_id = match event.get("sender") {
+                    Some(v) => match UserId::try_from(v.as_str()) {
+                        Ok(u) => u,
+                        Err(_) => return false,
+                    },
+                    None => return false,
+                };
+
+                let sender_level = match context.users_power_levels.get(&sender_id) {
+                    Some(i) => i,
+                    None => &context.default_power_level,
+                };
+
+                sender_level >= context.notification_power_levels.get(key)
+            }
+        }
+    }
+}
+
+/// The context of the room associated to an event to be able to test all push conditions
+#[derive(Clone, Debug)]
+#[cfg_attr(not(feature = "unstable-exhaustive-types"), non_exhaustive)]
```

Comment:
Yeah I wasn't sure about it either. It could be useful in case a new condition is added in the future and needs more context. But I don't mind removing that attribute

Reply:
But *needing* more context would mean a breaking change anyways. Let's remove the attribute.